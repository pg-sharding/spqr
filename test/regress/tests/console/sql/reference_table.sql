CREATE REFERENCE TABLE zz;

SYNC REFERENCE TABLES sh1;

DROP DISTRIBUTION ALL CASCADE;
DROP KEY RANGE ALL;
