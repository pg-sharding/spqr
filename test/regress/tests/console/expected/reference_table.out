
		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

CREATE REFERENCE TABLE zz;
   create reference table    
-----------------------------
 table    -> zz
 shard id -> sh1,sh2,sh3,sh4
(2 rows)

CREATE REFERENCE TABLE zz2 ON sh3, sh4;
 create reference table 
------------------------
 table    -> zz2
 shard id -> sh3,sh4
(2 rows)

CREATE REFERENCE RELATION zz2;
ERROR:  reference relation zz2 already exists
SYNC REFERENCE TABLES ON sh1;
ERROR:  request is unprocessable in router
ALTER DISTRIBUTION 'REPLICATED' ALTER RELATION zz2 SCHEMA my_schema;
          alter relation          
----------------------------------
 relation name   -> my_schema.zz2
 distribution id -> REPLICATED
(2 rows)

ALTER DISTRIBUTION 'REPLICATED' ALTER RELATION zz2 DISTRIBUTION KEY id;
ERROR:  setting distribution key is forbidden for reference relations
DROP REFERENCE RELATION zz2;
 drop reference table 
----------------------
 table -> zz2
(1 row)

--should fail
DROP REFERENCE RELATION zz2;
ERROR:  reference relation "zz2" not found
DROP DISTRIBUTION ALL CASCADE;
       drop distribution       
-------------------------------
 distribution id -> REPLICATED
(1 row)

DROP KEY RANGE ALL;
 drop key range 
----------------
(0 rows)

